yt/frontends/stream/fields.py

```diff
@@ -87,18 +89,43 @@ class StreamFieldInfo(FieldInfoContainer):
 
     def setup_fluid_fields(self):
         from yt.fields.magnetic_field import setup_magnetic_field_aliases
+        from yt.fields.species_fields import setup_species_fields
+        from yt.utilities.periodic_table import periodic_table
 
+        # First grab all the element symbols from the periodic table
+        symbols = list(periodic_table.elements_by_symbol.keys())
+        # Now add the electron, and some common molecules
+        symbols += ["El", "H2", "CO"]
+        species_names = []
         for field in self.ds.stream_handler.field_units:
             if field[0] in self.ds.particle_types:
                 continue
             units = self.ds.stream_handler.field_units[field]
             if units != "":
                 self.add_output_field(field, sampling_type="cell", units=units)
+                # Check to see if this could be a species fraction field
+                if field[1].endswith("_fraction"):
+                    sp = field[1].rsplit("_fraction")[0]
+                    parts = sp.split("_")
+                    # parts is now either an element or molecule symbol
+                    # by itself:
+                    valid = parts[0] in symbols
+                    # or it may also have an ionization state after it
+                    if len(parts) > 1 and parts[0] != "El":
+                        # Note that this doesn't catch invalid ionization states
+                        valid &= re.match("^[pm](0|[1-9][0-9]*)$", parts[1]) is not None
```

Comment:
just to make sure I'm reading this right: `"p0"`  is a valid ionisation expression ?

Reply:
Yes, `"p0"` is neutral.